src/api/app/views/webui/request/_add_role_request_dialog.html.haml

```diff
@@ -18,9 +18,24 @@
           .form-group
             = label_tag(:role, 'Add role:')
             = select_tag(:role, options_for_select([['Maintainer', 'maintainer'], ['Bugowner', 'bugowner']]), class: 'form-control')
-          .form-group
-            = label_tag(:user, 'For user:')
-            = text_field_tag(:user, User.session!.login, class: 'form-control')
+          .card
+            .bg-light
+              %ul.nav.nav-tabs.pt-2.px-3.flex-nowrap.collapsible{ 'role': 'tablist' }
+                %li.nav-item
+                  %a{ 'id' => 'user-tab', 'data-toggle' => 'tab', 'href' => '#user', 'class' => 'nav-link text-nowrap active',
+                   'role' => 'tab', 'aria-controls' => 'user', 'aria-selected' => 'true', 'onclick' => 'deleteGroupInput()' }
+                    For user
+                %li.nav-item
+                  %a{ 'id' => 'group-tab', 'data-toggle' => 'tab', 'href' => '#group', 'class' => 'nav-link text-nowrap',
+                   'role' => 'tab', 'aria-controls' => 'group', 'aria-selected' => 'false', 'onclick' => 'deleteUserInput()' }
+                    For groups
+              %div{ 'class' => 'tab-content' }
+                %div{ 'id' => 'user', 'class' => 'tab-pane fade show active', 'role' => 'tab', 'aria-labelledby' => 'user-tab' }
+                  = text_field_tag(:user, nil, id: 'user-text-field', class: 'form-control', placeholder: 'Enter username',
```

Comment:
As @krauselukas said, we have an autocomplete component which would be perfect for this PR. It's listed in our [patterns library](https://obs-patterns.netlify.com/molecules/autocomplete.html).

For a concrete example of how it's used for groups, have a look [here](https://github.com/openSUSE/open-build-service/blob/cc1d6484273fbb98bd2d96ca52917dd052e3bd97/src/api/app/views/webui/shared/user_or_groups_roles/_add_group_dialog.html.haml#L8-L9). For users, it's almost the same and you can check [here](https://github.com/openSUSE/open-build-service/blob/cc1d6484273fbb98bd2d96ca52917dd052e3bd97/src/api/app/views/webui/shared/user_or_groups_roles/_add_user_dialog.html.haml#L8-L9).